File: backend/app/schemas/analytics.py
from __future__ import annotations

from pydantic import BaseModel


class DashboardSummaryResponse(BaseModel):
    total_customers: int
    today_customers: int
    total_prescriptions: int
    bills_generated_today: int
    revenue_today: float
    scheduled_campaigns: int
    failed_whatsapp_jobs: int


class RevenueSummaryResponse(BaseModel):
    range_key: str
    total_revenue: float
    total_bills: int
    average_bill_value: float


class RevenueTimeseriesPoint(BaseModel):
    label: str
    value: float


class RevenueTimeseriesResponse(BaseModel):
    range_key: str
    points: list[RevenueTimeseriesPoint]

File: frontend/src/types/analytics.ts
export type RevenueRange = "today" | "last_7_days" | "last_30_days";

export type DashboardSummary = {
  total_customers: number;
  today_customers: number;
  total_prescriptions: number;
  bills_generated_today: number;
  revenue_today: number;
  scheduled_campaigns: number;
  failed_whatsapp_jobs: number;
};

export type RevenueSummary = {
  range_key: RevenueRange;
  total_revenue: number;
  total_bills: number;
  average_bill_value: number;
};

export type RevenueTimeseriesPoint = {
  label: string;
  value: number;
};

export type RevenueTimeseries = {
  range_key: RevenueRange;
  points: RevenueTimeseriesPoint[];
};
